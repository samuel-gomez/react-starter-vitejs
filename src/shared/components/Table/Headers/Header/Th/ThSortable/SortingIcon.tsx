import { NONE, ASCENDING, DESCENDING } from 'shared/components/Table/constants';
import type { Torder } from './ThSortable.container';

export const orderIcons = (order: Torder = NONE) =>
  ({
    [NONE]: 'sorting',
    [ASCENDING]: 'arrow-xs-up',
    [DESCENDING]: 'arrow-xs-down',
  })[order];

export type TSortingIcon = {
  orderIconsFn?: typeof orderIcons;
  order?: Torder;
};

const SortingIcon = ({ order, orderIconsFn = orderIcons }: TSortingIcon) => (
  <span className={`af-btn__icon af-btn__icon--table-sorting glyphicon glyphicon-${orderIconsFn(order)}`} />
);

export default SortingIcon;
