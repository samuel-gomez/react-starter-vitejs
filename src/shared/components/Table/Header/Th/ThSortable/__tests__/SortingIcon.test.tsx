import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import SortingIcon, { orderIcons } from '../SortingIcon';

describe('orderIcons', () => {
  it.each`
    order        | expected
    ${undefined} | ${'sorting'}
    ${'NONE'}    | ${'sorting'}
    ${1}         | ${'arrow-xs-up'}
    ${-1}        | ${'arrow-xs-down'}
  `('Should return $expected when orderIcons have been called with order: $order', ({ order, expected }) => {
    const result = orderIcons(order);

    expect(result).toEqual(expected);
  });
});

describe('SortingIcon', () => {
  it('Should render SortingIcon with order equal NONE', () => {
    const { asFragment } = render(<SortingIcon />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should render SortingIcon with order equal 1', () => {
    const { asFragment } = render(<SortingIcon order={1} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('Should render SortingIcon with order equal -1', () => {
    const { asFragment } = render(<SortingIcon order={-1} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
