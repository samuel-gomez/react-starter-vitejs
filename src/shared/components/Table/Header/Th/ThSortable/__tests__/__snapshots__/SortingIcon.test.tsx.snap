// Vitest Snapshot v1

exports[`SortingIcon > Should render SortingIcon with order equal -1 1`] = `
<DocumentFragment>
  <span
    class="af-btn__icon af-btn__icon--table-sorting glyphicon glyphicon-arrow-xs-down"
  />
</DocumentFragment>
`;

exports[`SortingIcon > Should render SortingIcon with order equal 1 1`] = `
<DocumentFragment>
  <span
    class="af-btn__icon af-btn__icon--table-sorting glyphicon glyphicon-arrow-xs-up"
  />
</DocumentFragment>
`;

exports[`SortingIcon > Should render SortingIcon with order equal NONE 1`] = `
<DocumentFragment>
  <span
    class="af-btn__icon af-btn__icon--table-sorting glyphicon glyphicon-sorting"
  />
</DocumentFragment>
`;
